init/action.yml

```
@@ -38,6 +38,9 @@ inputs:
     description: Try to auto-install your python dependencies
     required: true
     default: 'true'
+  source-root:
+    description: Path to the root source-code directory, relative to the workspace.
```

```suggestion
    description: Path of the root source code directory, relative to $GITHUB_WORKSPACE.
```

---

If we use `path.resolve()`, as you suggested above, perhaps it should just be
```suggestion
    description: Path of the root source code directory.
```

---

I would still leave in the `relative to $GITHUB_WORKSPACE` bit to make it clear how relative paths are interpreted. With `resolve` the path is still interpreted relative to the workspace. The main difference is how absolute paths are treated. `join` would append the workspace and the absolute path, while `resolve` returns just the absolute path.

So if you specify:
```
  source-root: foo/bar
```
You end up with
* `join` : `$GITHUB_WORKSPACE/foo/bar`
* `resolve`: `$GITHUB_WORKSPACE/foo/bar`

and if you specify:
```
  source-root: ${{ github.workspace }}/foo/bar
```
You end up with:
* `join` : `$GITHUB_WORKSPACE/$GITHUB_WORKSPACE/foo/bar`
* `resolve`: `$GITHUB_WORKSPACE/foo/bar`

I think the behaviour of `resolve` is more likely "what the user intended".